datascan/terraform/dataplex_data_quality.tf

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_dataplex_datascan" "data_quality_scan" {
+  location     = var.region
+  project      = var.project_id
+  data_scan_id = "template-dq-scan"
+  labels = {}
+
+  data {
+    resource = "//bigquery.googleapis.com/projects/${var.project_id}/datasets/{DATASET-ID}/tables/{TABLE}"
+  }
+
+  execution_spec {
+    trigger {
+      on_demand {}
+    }
+  }
+
+  # Custom logic to parse out rules metadata from a local rules.yaml file
+  data_quality_spec {
+    sampling_percent = try(local.sampling_percent, null)
+    row_filter       = try(local.row_filter, null)
+
+    dynamic "rules" {
```

Comment:
It would be a good idea to add support for new DQ Rule, SQLAssertion: https://source.corp.google.com/piper///depot/google3/google/cloud/dataplex/v1/data_quality.proto;l=339;bpv=1;bpt=1.